### PR TITLE
rust: eclass cleanups, mips64el, loong, 1.71.0

### PR DESCRIPTION
**Pull request by xen0n**

### CODE REVIEW

**Comment by gyakovlev** [comment]
Ack, will look closer soon and merge. Needs rebased for 1.70.0 though which just landed.

btw, while you at it, can you also start looking at conditionally adding loong definitions there. Needed for adding loong as we discussed on irc.
Eclass should not emit loong uri for versions before 1.71.0 for example.

**Comment by ionenwks** [comment]
```
Build completed successfully in 0:01:15
rm: cannot remove '/var/tmp/portage/dev-lang/rust-1.71.0/image/usr/lib/rust/1.71.0/etc/bash_completion.d/cargo': No such file or directory
 * ERROR: dev-lang/rust-1.71.0::gentoo failed (install phase):
 *   (no error message)
```
```
[ebuild     U  ] dev-lang/rust-1.71.0:stable/1.71::gentoo [1.70.0:stable/1.70::gentoo] USE="lto system-bootstrap system-llvm (-big-endian) -clippy -debug -dist -doc (-llvm-libunwind) (-miri) -nightly (-parallel-compiler) -profiler -rust-analyzer -rust-src -rustfmt -test -verify-sig -wasm" ABI_X86="32 (64) (-x32)" CPU_FLAGS_X86="sse2"
```

File: dev-lang/rust/rust-1.71.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+	fi
+}
+
+src_install() {
+	DESTDIR="${D}" "${EPYTHON}" ./x.py install -vv --config="${S}"/config.toml -j$(makeopts_jobs) || die
+
+	# bug #689562, #689160
+	rm -v "${ED}/usr/lib/${PN}/${PV}/etc/bash_completion.d/cargo" || die
+	rmdir -v "${ED}/usr/lib/${PN}/${PV}"/etc{/bash_completion.d,} || die
+	newbashcomp src/tools/cargo/src/etc/cargo.bashcomp.sh cargo
```
Comment (ionenwks) [inline dev-lang/rust/rust-1.71.0.ebuild:636]: wrt previous failure, wonder if they typo'ed the path while changing the style? (from etc/ to src/etc/) -- not that I can confirm what this is doing yet
```diff
--- rustc-1.70.0-src/src/bootstrap/dist.rs	2023-05-31 15:28:10.000000000 -0400
+++ rustc-1.71.0-src/src/bootstrap/dist.rs	2023-07-11 23:32:40.000000000 -0400
@@ -1066,7 +1071,11 @@
 
         tarball.add_file(&cargo, "bin", 0o755);
         tarball.add_file(etc.join("_cargo"), "share/zsh/site-functions", 0o644);
-        tarball.add_renamed_file(etc.join("cargo.bashcomp.sh"), "etc/bash_completion.d", "cargo");
+        tarball.add_renamed_file(
+            etc.join("cargo.bashcomp.sh"),
+            "src/etc/bash_completion.d",
+            "cargo",
+        );
         tarball.add_dir(etc.join("man"), "share/man/man1");
         tarball.add_legal_and_readme_to("share/doc/cargo");
 ```

Comment (xen0n) [reply dev-lang/rust/rust-1.71.0.ebuild:636]: For the record: the upstream change is https://github.com/rust-lang/rust/pull/110906.

Comment (ionenwks) [reply dev-lang/rust/rust-1.71.0.ebuild:636]: I see, so it was intentional.. albeit looks like it got reverted. Revert is just not in this version.

https://github.com/rust-lang/rust/commit/05bc71cfaf9eb35caab425126d4fa8dcb8c62072

Comment (xen0n) [reply dev-lang/rust/rust-1.71.0.ebuild:636]: > [rust-lang/rust@05bc71c](https://github.com/rust-lang/rust/commit/05bc71cfaf9eb35caab425126d4fa8dcb8c62072)

Corresponding PR: https://github.com/rust-lang/rust/pull/113579

Multiple parties waiting for a stable backport it seems. Probably best to wait until it's resolved upstream, but we can also just integrate that revert ourselves...

Comment (thesamesam) [reply dev-lang/rust/rust-1.71.0.ebuild:636]: Yeah, I wouldn't bother waiting. It's both trivial and upstream also don't seem to do this sort of thing very proactively either.

File: dev-lang/rust-bin/rust-bin-1.71.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+HOMEPAGE="https://www.rust-lang.org/"
+SRC_URI="$(rust_all_arch_uris ${MY_P})
+	rust-src? ( ${MY_SRC_URI} )
+"
+# Keep this separate to allow easy commenting out if not yet built
+#SRC_URI+=" sparc? ( ${GENTOO_BIN_BASEURI}/${MY_P}-sparc64-unknown-linux-gnu.tar.xz -> ${MY_P}-sparc64-unknown-linux-gnu.sam.tar.xz ) "
+
+LICENSE="|| ( MIT Apache-2.0 ) BSD BSD-1 BSD-2 BSD-4 UoI-NCSA"
+SLOT="stable"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86"
```
Comment (thesamesam) [inline dev-lang/rust-bin/rust-bin-1.71.0.ebuild:23]: Drop ~sparc until I can build the bin

**Review by thesamesam — APPROVED** [review]

File: dev-lang/rust/rust-1.71.0.ebuild (hunk 1 above)
Comment (xen0n) [reply dev-lang/rust/rust-1.71.0.ebuild:637]: This is now integrated.

File: dev-lang/rust-bin/rust-bin-1.71.0.ebuild (hunk 2 above)
Comment (xen0n) [reply dev-lang/rust-bin/rust-bin-1.71.0.ebuild]: done

**Review by ionenwks — APPROVED** [review]

**Pull request closed without merging**